crates/ika-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -150,9 +155,27 @@ impl DWalletMPCManager {
             malicious_handler: MaliciousHandler::new(epoch_store.committee().clone()),
             pending_for_computation_order: VecDeque::new(),
             pending_for_events_order: Default::default(),
+            last_completed_session_sequence_number: 0,
         })
     }
 
+    pub(crate) fn update_last_completed_session_sequence_number(
+        &mut self,
+        last_completed_session_sequence_number: u64,
+    ) {
+        if last_completed_session_sequence_number < self.last_completed_session_sequence_number {
+            return;
+        }
+        for i in self.last_completed_session_sequence_number
+            ..=(last_completed_session_sequence_number + MAX_ACTIVE_SESSIONS_BUFFER)
+        {
+            if let Some(session) = self.pending_sessions.remove(&i) {
+                self.mpc_sessions.insert(session.session_id, session);
+            }
+        }
```

Comment:
It is anyways not how we discussed as far as I know: the Move should keep a `last_event_to_execute` and not be computed here. We should think if this ,akes any difference